demo/c/koala_demo_file.c

```diff
@@ -0,0 +1,341 @@
+                library_path = optarg;
+                break;
+            case 'm':
+                model_path = optarg;
+                break;
+            case 'a':
+                access_key = optarg;
+                break;
+            case 'i':
+                input_path = optarg;
+                break;
+            case 'o':
+                output_path = optarg;
+                break;
+            default:
+                exit(EXIT_FAILURE);
+        }
+    }
+
+    if (!library_path || !access_key || !input_path || !output_path) {
+        print_usage(argv[0]);
+        exit(EXIT_FAILURE);
+    }
+
+    void *koala_library = open_dl(library_path);
+    if (!koala_library) {
+        fprintf(stderr, "failed to open library at '%s'.\n", library_path);
+        exit(EXIT_FAILURE);
+    }
+
+    const char *(*pv_status_to_string_func)(pv_status_t) =
+            load_symbol(koala_library, "pv_status_to_string");
+    if (!pv_status_to_string_func) {
+        print_dl_error("failed to load 'pv_status_to_string'");
+        exit(EXIT_FAILURE);
+    }
+
+    int32_t(*pv_sample_rate_func)() = load_symbol(koala_library, "pv_sample_rate");
+    if (!pv_sample_rate_func) {
+        print_dl_error("failed to load 'pv_sample_rate'");
+        exit(EXIT_FAILURE);
+    }
+
+    pv_status_t(*pv_koala_init_func)(const char *, const char *, pv_koala_t * *) =
+            load_symbol(koala_library, "pv_koala_init");
+    if (!pv_koala_init_func) {
+        print_dl_error("failed to load 'pv_koala_init'");
+        exit(EXIT_FAILURE);
+    }
+
+    void (*pv_koala_delete_func)(pv_koala_t *) = load_symbol(koala_library, "pv_koala_delete");
+    if (!pv_koala_delete_func) {
+        print_dl_error("failed to load 'pv_koala_delete'");
+        exit(EXIT_FAILURE);
+    }
+
+    pv_status_t(*pv_koala_process_func)(pv_koala_t * , const int16_t *, int16_t *) =
+            load_symbol(koala_library, "pv_koala_process");
+    if (!pv_koala_process_func) {
+        print_dl_error("failed to load 'pv_koala_process'");
+        exit(EXIT_FAILURE);
+    }
+
+    int32_t(*pv_koala_frame_length_func)() = load_symbol(koala_library, "pv_koala_frame_length");
+    if (!pv_koala_frame_length_func) {
+        print_dl_error("failed to load 'pv_koala_frame_length'");
+        exit(EXIT_FAILURE);
+    }
+
+    const char *(*pv_koala_version_func)() = load_symbol(koala_library, "pv_koala_version");
+    if (!pv_koala_version_func) {
+        print_dl_error("failed to load 'pv_koala_version'");
+        exit(EXIT_FAILURE);
+    }
+
+    pv_koala_t *koala = NULL;
+    pv_status_t koala_status = pv_koala_init_func(access_key, model_path, &koala);
+    if (koala_status != PV_STATUS_SUCCESS) {
+        fprintf(stderr, "failed to init with '%s'", pv_status_to_string_func(koala_status));
+        exit(EXIT_FAILURE);
+    }
+    fprintf(stdout, "V%s\n\n", pv_koala_version_func());
+
+    drwav input_file;
+
+    if (!drwav_init_file(&input_file, input_path, NULL)) {
```

Comment:
Some additional changes are needed to support funky audio file paths on windows, see this PR for details: https://github.com/Picovoice/porcupine/pull/890/commits/394209aacf9cf819afa1afb9738338ac73494125

Reply:
good point!